hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -126,66 +135,146 @@ def accept(self, settings, item):
+            # the shot, but there might be cases where this heads off a problem.
+            publish_entity_type = get_published_file_entity_type(publisher.sgtk)
+
+            self.logger.debug("Clip publish types to search for: %s" % CLIP_PUBLISH_TYPE)
+
+            clip_publishes = publisher.shotgun.find(
+                publish_entity_type,
+                [
+                    ["entity", "is", item.context.entity],
+                    ["published_file_type.PublishedFileType.code", "is", CLIP_PUBLISH_TYPE],
+                ],
+                fields=(
+                    "path",
+                    "version_number",
+                    "name",
+                    "published_file_type",
+                    "description",
+                ),
+                order=[dict(field_name="id", direction="desc")],
+            )
+
+            if clip_publishes:
+                self.logger.debug("Found published clips: %s" % clip_publishes)
+
+            for clip_publish in clip_publishes:
+                self.logger.debug("Checking existence of OpenClip: %s" % clip_publish)
+
+                try:
+                    flame_clip_path = resolve_publish_path(publisher.sgtk, clip_publish)
+                except Exception:
+                    self.logger.debug("Unable to resolve path: %s" % clip_publish)
+                    continue
+
+                if os.path.exists(flame_clip_path):
+                    # If we already found a newer clip that we're going to use, we
+                    # log the fact that the others will NOT be used for the sake of
+                    # transparency and debuggability.
+                    if item.properties.get("flame_clip_path"):
+                        self.logger.warning(
+                            "The following clip exists, but will not be updated since a "
+                            "newer publish exists: %s" % flame_clip_path
+                        )
+                        continue
+                    else:
+                        self.logger.info("Clip publish found: %s" % flame_clip_path)
+
+                    # Keep track of the PublishedFile entity. We'll need it
+                    # later when the clip is updated, at which time we'll
+                    # be versioning up the PublishedFile.
+                    item.properties["flame_clip_publish"] = clip_publish
+                    item.properties["flame_clip_path"] = flame_clip_path
+                else:
+                    self.logger.debug(
+                        "Published clip file isn't accessible: %s" % flame_clip_path
+                    )
         else:
             self.logger.debug(
-                "No flame clip template configured. "
-                "Plugin %s no accepting item: %s" % (self, item)
+                "Unable to look up a clip file to update. No template exists, "
+                "and the item's context is not associated with an entity. This "
```

Comment:
nit: the part of the comment speculating about the project context is perhaps not necessary?